### PR TITLE
Join cli+keymanager account managment

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

Current unstable has some serious issues:
- Keymanager does not persist keystores in a way that are immediately runnable, rendering the feature not usable
- There are multiple code paths to import keystores, between import cmd, validator cmd, and keymanager. This inconsistencies are confusing, and have caused issues in the past.
- lockfile logic is dangerously scattered across multiple code paths, which can lead to the lock not being detected.
- RemoteKey server is incomplete pending this PR

**Description**

Join cli+keymanager account managment into cli

- Move keymanager code into validator cli
- keystores are _only_ written and read to disk via PersistedKeysBackend. import cmd, validator cmd and keymanager all use this class
- Make keymanager persisted keystores and remoteKeys readable after a restart

### CODE REVIEW

**Comment by g11tech** [comment]
structure looks good! :+1:

File: packages/cli/src/cmds/validator/keymanager/persistedKeys.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    passphraseFilepath: string;
+  } {
+    // TODO: Ensure correct formating 0x prefixed
+
+    const dirpath = path.join(this.paths.keystoresDir, pubkeyHex);
+
+    return {
+      dirpath,
+      keystoreFilepath: path.join(this.paths.secretsDir, pubkeyHex),
+      passphraseFilepath: path.join(dirpath, "voting-keystore.json"),
```
Comment (wemeetagain) [inline packages/cli/src/cmds/validator/keymanager/persistedKeys.ts]: are these swapped?

Comment (dapplion) [reply packages/cli/src/cmds/validator/keymanager/persistedKeys.ts]: Yep, good pick

**Comment by dapplion** [comment]
Last commits add e2e tests that programatically tests this flows:
- run validation directly with keystores in disk
- import keystores from disk then run
- import keystores from API then run
- import remote keys from API then run

Plus
- Delete keys flows
- Test .lock if run twice
- Test authentication

**Comment by dapplion** [comment]
Test this build's cli manually to ensure validators are read correctly :+1:

**Review by wemeetagain — APPROVED** [review]

File: packages/cli/src/util/passphrase.ts [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 
 /**
  * Utility to read file as UTF8 and strip any trailing new lines.
  * All passphrase files must be read with this function
  */
 export function readPassphraseFile(passphraseFile: string): string {
   const data = fs.readFileSync(passphraseFile, "utf8");
-  const passphrase = stripOffNewlines(data);
+  // Remove trailing new lines '\n' or '\r' if any
+  const passphrase = data.replace(/[\n|\r]+$/g, "");
```
Comment (nflaig) [inline packages/cli/src/util/passphrase.ts:10]: @dapplion  what was the reason for changing this line?

I would recommend that we clean this up, either use `stripOffNewlines` here or get rid of the function altogether

https://github.com/ChainSafe/lodestar/blob/9f7b83037978ceaba49119b9506567078d12fdec/packages/cli/src/util/stripOffNewlines.ts#L4

Comment (dapplion) [reply packages/cli/src/util/passphrase.ts:10]: I don't remember to be honest

Comment (nflaig) [reply packages/cli/src/util/passphrase.ts:10]: let's clean this up https://github.com/ChainSafe/lodestar/pull/5283

**Pull request merged**